internal/mode/static/state/graph/gateway_listener.go

```diff
@@ -148,40 +150,45 @@ type listenerConfigurator struct {
 func (c *listenerConfigurator) configure(listener v1.Listener) *Listener {
 	var conds []conditions.Condition
 
+	attachable := true
+
 	// validators might return different conditions, so we run them all.
 	for _, validator := range c.validators {
-		conds = append(conds, validator(listener)...)
+		currConds, currAttachable := validator(listener)
+		conds = append(conds, currConds...)
+
+		attachable = attachable && currAttachable
 	}
 
+	valid := len(conds) == 0
+
 	var allowedRouteSelector labels.Selector
 	if selector := GetAllowedRouteLabelSelector(listener); selector != nil {
 		var err error
 		allowedRouteSelector, err = metav1.LabelSelectorAsSelector(selector)
 		if err != nil {
 			msg := fmt.Sprintf("invalid label selector: %s", err.Error())
 			conds = append(conds, staticConds.NewListenerUnsupportedValue(msg)...)
+			valid = false
 		}
 	}
 
 	supportedKinds := getListenerSupportedKinds(listener)
 
-	if len(conds) > 0 {
-		return &Listener{
-			Source:         listener,
-			Conditions:     conds,
-			Valid:          false,
-			SupportedKinds: supportedKinds,
-		}
-	}
-
```

Comment:
Not sure I see the benefit of this change. Couldn't you just add the `attachable` bool to this returned value, and the Listener initialized after this, and that's all you need to change?

The following Listener you add the conditions, but we know that those conditions are empty.